lua/nvim-tree/explorer/sorters.lua

```diff
@@ -95,67 +86,16 @@ local function split_merge(t, first, last, comparator)
   merge(t, first, mid, last, comparator)
 end
 
----Perform a merge sort using sorter option.
----@param t table nodes
-function M.sort(t)
-  if C.user then
-    local t_user = {}
-    local origin_index = {}
-
-    for _, n in ipairs(t) do
-      table.insert(t_user, {
-        absolute_path = n.absolute_path,
-        executable = n.executable,
-        extension = n.extension,
-        filetype = vim.filetype.match { filename = n.name },
-        link_to = n.link_to,
-        name = n.name,
-        type = n.type,
-      })
-      table.insert(origin_index, n)
-    end
-
-    local predefined = C.user(t_user)
-    if predefined then
-      split_merge(t, 1, #t, get_comparator(predefined))
-      return
-    end
-
-    -- do merge sort for prevent memory exceed
-    local user_index = {}
-    for i, v in ipairs(t_user) do
-      if type(v.absolute_path) == "string" and user_index[v.absolute_path] == nil then
-        user_index[v.absolute_path] = i
-      end
-    end
-
-    -- if missing value found, then using origin_index
-    local mini_comparator = function(a, b)
-      local a_index = user_index[a.absolute_path] or origin_index[a.absolute_path]
-      local b_index = user_index[b.absolute_path] or origin_index[b.absolute_path]
-
-      if type(a_index) == "number" and type(b_index) == "number" then
-        return a_index <= b_index
-      end
-      return (a_index or 0) <= (b_index or 0)
-    end
-
-    split_merge(t, 1, #t, mini_comparator) -- sort by user order
-  else
-    split_merge(t, 1, #t, get_comparator(M.config.sort.sorter))
-  end
-end
-
```

Comment:
I know this is not "right" however I'd prefer not to pollute git history with these refactors.

Could you possibly reorder this one?

gunzip [reorder.patch.gz](https://github.com/user-attachments/files/16211123/reorder.patch.gz)
git apply reorder.patch

Reply:
Indeed, now it's more visible what exactly was changed :)